### PR TITLE
Upgrade fs2 to 0.10.0-M9

### PR DESCRIPTION
**Pull request by aeons**
Everything compiles, but tests needs updated jawn-fs2 and fs2-reactive-streams.

- [x] [jawn-fs2](https://github.com/http4s/jawn-fs2/pull/13)
- [x] [fs2-reactive-streams](https://github.com/zainab-ali/fs2-reactive-streams/pull/40)

### CODE REVIEW

**Comment by lukestephenson** [comment]
This looks good to me.  I agree with your comment on gitter:
> there’s a lot of Chunk[Byte] around that are from before segment was introduced
since we rarely do index-based access, I think those should be Segment[Byte, Unit] instead

**Comment by rossabaker** [comment]
jawn-fs2-0.12.0-M4 is released.

File: blaze-core/src/main/scala/org/http4s/blazecore/Http1Stage.scala [hunk 1]
```diff
@@ -75,7 +75,7 @@ trait Http1Stage[F[_]] { self: TailStage[ByteBuffer] =>
       rr: StringWriter,
       minor: Int,
       closeOnFinish: Boolean): Http1Writer[F] = lengthHeader match {
-    case Some(h) if bodyEncoding.map(!_.hasChunked).getOrElse(true) || minor == 0 =>
+    case Some(h) if bodyEncoding.forall(!_.hasChunked) || minor == 0 =>
```
Comment (rossabaker) [inline blaze-core/src/main/scala/org/http4s/blazecore/Http1Stage.scala]: Good use of `forall`!

File: blaze-core/src/main/scala/org/http4s/blazecore/util/IdentityWriter.scala [hunk 2]
```diff
@@ -12,18 +12,18 @@ import org.http4s.util.chunk._
 import org.log4s.getLogger
 import scala.concurrent.{ExecutionContext, Future}
 
-private[http4s] class IdentityWriter[F[_]](size: Long, out: TailStage[ByteBuffer])(
+private[http4s] class IdentityWriter[F[_]](size: Int, out: TailStage[ByteBuffer])(
```
Comment (rossabaker) [inline blaze-core/src/main/scala/org/http4s/blazecore/util/IdentityWriter.scala]: What forced this back to `Int`?

File: core/src/main/scala/org/http4s/util/Renderable.scala [hunk 3]
```diff
@@ -197,7 +197,8 @@ final case class ChunkWriter(
 ) extends Writer {
 
   override def append(s: String): this.type = {
-    toChunk = (toChunk ++ Chunk.bytes(s.getBytes(charset))).toChunk
+    // This smells wrong
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/util/Renderable.scala]: Is it better for this to be a `SegmentWriter`?

File: blaze-core/src/main/scala/org/http4s/blazecore/Http1Stage.scala (hunk 1 above)
Comment (aeons) [reply blaze-core/src/main/scala/org/http4s/blazecore/Http1Stage.scala]: That was an autofix suggestion by IntelliJ :)

File: blaze-core/src/main/scala/org/http4s/blazecore/util/IdentityWriter.scala (hunk 2 above)
Comment (aeons) [reply blaze-core/src/main/scala/org/http4s/blazecore/util/IdentityWriter.scala]: `take` on `Chunk` is Int, while on `Segment` it's Long. I think before it it used the inherited take from Segment.

File: core/src/main/scala/org/http4s/util/Renderable.scala (hunk 3 above)
Comment (aeons) [reply core/src/main/scala/org/http4s/util/Renderable.scala]: Yes, I think so, but I think that would be part of a larger overhaul to use Segment generally.

**Pull request merged**